README.md

```diff
@@ -101,6 +101,25 @@ const saml = new SAML(options);
 - `additionalLogoutParams`: dictionary of additional query params to add to 'logout' requests
 - `logoutCallbackUrl`: The value with which to populate the `Location` attribute in the `SingleLogoutService` elements in the generated service provider metadata.
 
+- **SAML Extensions**
+- `samlExtensions`: Optional, The SAML extension provides a more flexible structure for expressing which combination of Attributes are requested by service providers in comparison to the existing mechanisms, [More about extensions](https://docs.oasis-open.org/security/saml-protoc-req-attr-req/v1.0/saml-protoc-req-attr-req-v1.0.html). There are many possible values for the `samlExtensions` element. It accepts fully customize [XMLBuilder](https://www.npmjs.com/package/xmlbuilder) type.
```

Comment:
Since this PR is about enabling `Extensions` to various requests (which extend `samlp:RequestAbstractType`) supported by `node-saml` there is also `LogoutRequest` which may contain request specific `Extensions`.

```xml
…
    <complexType name="RequestAbstractType" abstract="true">                                                 
        <sequence>                                                                                           
            <element ref="saml:Issuer" minOccurs="0"/>                                                       
            <element ref="ds:Signature" minOccurs="0"/>                                                      
            <element ref="samlp:Extensions" minOccurs="0"/>                                                  
        </sequence>                                                                                          
        <attribute name="ID" type="ID" use="required"/>                                                      
        <attribute name="Version" type="string" use="required"/>                                             
        <attribute name="IssueInstant" type="dateTime" use="required"/>                                      
        <attribute name="Destination" type="anyURI" use="optional"/>                                         
        <attribute name="Consent" type="anyURI" use="optional"/>                                             
    </complexType>
…
    <complexType name="LogoutRequestType">                                                                   
        <complexContent>                                                                                     
            <extension base="samlp:RequestAbstractType">                                                     
                <sequence>                                                                                   
                    <choice>                                                                                 
                        <element ref="saml:BaseID"/>                                                         
                        <element ref="saml:NameID"/>                                                         
                        <element ref="saml:EncryptedID"/>                                                    
                    </choice>                                                                                
                    <element ref="samlp:SessionIndex" minOccurs="0" maxOccurs="unbounded"/>                  
                </sequence>                                                                                  
                <attribute name="Reason" type="string" use="optional"/>                                      
                <attribute name="NotOnOrAfter" type="dateTime" use="optional"/>                              
            </extension>                                                                                     
        </complexContent>                                                                                    
    </complexType>
…
```
source: https://docs.oasis-open.org/security/saml/v2.0/saml-schema-protocol-2.0.xsd

PR's current implementation skip/ignore `LogoutRequest`

Reply:
nice catch @srd90 , we will add it in another PR

Reply:
Another PR, or in another push to this PR?

Reply:
I think another PR

Reply:
Just to be thorough, this [comment](https://github.com/node-saml/node-saml/pull/11/files#r683201959) indicates that you'll do another PR for the `LogoutRequest` version of Extensions. Does that mean that we should be using a more descriptive name instead of `samlExtensions` so there isn't a future name collision for that next PR?

Reply:
I think right. we should make it more descriptive like `samlAuthnRequestExtensions`, what do you think @cjbarth ?